bin/stress-test/src/main.rs

```diff
@@ -0,0 +1,188 @@
+    task::spawn(async move { store.serve().await.context("Serving store") });
+
+    // Start block builder
+    let store_client =
+        StoreClient::new(ApiClient::connect(store_config.endpoint.to_string()).await.unwrap());
+    let block_builder = BlockBuilder::new(store_client);
+
+    println!("Creating new faucet account...");
+    let coin_seed: [u64; 4] = rand::thread_rng().gen();
+    let mut rng = RpoRandomCoin::new(coin_seed.map(Felt::new));
+    let key_pair = SecretKey::with_rng(&mut rng);
+    let init_seed = [0_u8; 32];
+    let (new_faucet, _seed) = AccountBuilder::new(init_seed)
+        .anchor(AccountIdAnchor::PRE_GENESIS)
+        .account_type(AccountType::FungibleFaucet)
+        .storage_mode(AccountStorageMode::Private)
+        .with_component(RpoFalcon512::new(key_pair.public_key()))
+        .with_component(
+            BasicFungibleFaucet::new(TokenSymbol::new("TEST").unwrap(), 2, Felt::new(100_000))
+                .unwrap(),
+        )
+        .build()
+        .unwrap();
+    let faucet_id = new_faucet.id();
+
+    let (batch_sender, batch_receiver) = channel::<TransactionBatch>();
+
+    println!("Inserting blocks...");
+    let load_start = Instant::now();
+
+    // Spawn a task for block building
+    let db_task = task::spawn(async move {
+        let mut current_block: Vec<TransactionBatch> = Vec::with_capacity(BATCHES_PER_BLOCK);
+        let mut insertion_times = Vec::new();
+        while let Ok(batch) = batch_receiver.recv() {
+            current_block.push(batch);
+
+            if current_block.len() == BATCHES_PER_BLOCK {
+                let start = Instant::now();
+                block_builder.build_block(&current_block).await.unwrap();
+                insertion_times.push(start.elapsed());
+                current_block.clear();
+            }
+        }
+
+        if !current_block.is_empty() {
+            let start = Instant::now();
+            block_builder.build_block(&current_block).await.unwrap();
+            insertion_times.push(start.elapsed());
+        }
+
+        // Print insertion times
+        let insertions = insertion_times.len() as u32;
+        println!("Inserted {} blocks", insertions);
+
+        if insertions == 0 {
+            return;
+        }
+
+        let total_time: Duration = insertion_times.iter().sum();
+        let avg_time = total_time / insertions;
+        println!("Average insertion time: {} ms", avg_time.as_millis());
+    });
+
+    // Parallel account grinding and batch generation
+    (0..accounts_number)
+        .into_par_iter()
+        .map(|_| {
+            let (new_account, _) = AccountBuilder::new(init_seed)
+                .anchor(AccountIdAnchor::PRE_GENESIS)
+                .account_type(AccountType::RegularAccountImmutableCode)
+                .storage_mode(AccountStorageMode::Private)
+                .with_component(RpoFalcon512::new(key_pair.public_key()))
+                .with_component(BasicWallet)
```

Comment:
I'd make a few changes here:

1. Use genesis block as the anchor (rather than pre-genesis).
2. Use a different `init_seed` for each account.

We could also use a different signature key for each account - but Falcon key generation is expensive (would take almost 0.5 seconds). So, re-using the keys as long as the seeds are different should be fine.

Also, separate from this PR, I'm curious why inserting identical accounts into the store didn't trigger any errors. Maybe something for us to take a look at in the future. cc @Mirko-von-Leipzig.

Reply:
Created #629 to look into why.